src/NHibernate/Util/ArrayHelper.cs

```diff
@@ -114,6 +114,13 @@ public static void AddAll<T>(IList<T> to, IList<T> from)
 				to.Add(obj);
 		}
 
+		internal static void AddAll<T>(List<T> to, IList from)
+		{
```

Comment:
To have performance gain we need `from` to be ICollection\<T\>. But in our case `from` is  List\<object\>. So `Cast` will lead to worse performance then original implementation.

I've updated code to avoid second data copy by adding generic wrapper over 'ICollection'